File: SproInstall/SocketPro/tutorial/CSharp/MySecure/MySecureClient/MyBlowFish.cs


using System;
using SocketProAdapter;
using SocketProAdapter.ClientSide;
using USOCKETLib;

public class CMySecure : CAsyncServiceHandler
{
    public CMySecure(CClientSocket cs, IAsyncResultsHandler pDefaultAsyncResultsHandler)
        : base(MyBlowFishConst.sidCMySecure, cs, pDefaultAsyncResultsHandler)
    {
    }

    public CMySecure(CClientSocket cs)
        : base(MyBlowFishConst.sidCMySecure, cs)
    {
    }

    public CMySecure()
        : base(MyBlowFishConst.sidCMySecure)
    {
    }

    public int m_nErrorCode = 0;
    public string m_strErrorMessage;

	protected string m_OpenRtn;
	protected void OpenAsyn(string strUserIDToDB, string strPasswordToDB)
	{
        SendRequest(MyBlowFishConst.idOpenCMySecure, strUserIDToDB, strPasswordToDB);
	}

	protected bool m_BeginTransRtn;
	public void BeginTransAsyn()
	{
		SendRequest(MyBlowFishConst.idBeginTransCMySecure);
	}

	protected bool m_ExecuteNoQueryRtn;
	public void ExecuteNoQueryAsyn(string strSQL)
	{
		SendRequest(MyBlowFishConst.idExecuteNoQueryCMySecure, strSQL);
	}

	protected bool m_CommitRtn;
	public void CommitAsyn(bool bSmart)
	{
		SendRequest(MyBlowFishConst.idCommitCMySecure, bSmart);
	}

	//When a result comes from a remote SocketPro server, the below virtual function will be called.
	//We always process returning results inside the function.
	protected override void OnResultReturned(short sRequestID, CUQueue UQueue)
	{
        UQueue.Pop(out m_nErrorCode);
        UQueue.Load(out m_strErrorMessage);

		switch(sRequestID)
		{
		case MyBlowFishConst.idOpenCMySecure:
			UQueue.Load(out m_OpenRtn);
			break;
		case MyBlowFishConst.idBeginTransCMySecure:
			UQueue.Pop(out m_BeginTransRtn);
			break;
		case MyBlowFishConst.idExecuteNoQueryCMySecure:
			UQueue.Pop(out m_ExecuteNoQueryRtn);
			break;
		case MyBlowFishConst.idCommitCMySecure:
			UQueue.Pop(out m_CommitRtn);
			break;
		default:
			break;
		}
	}
	public string Open(string strUserIDToDB, string strPasswordToDB)
	{
		OpenAsyn(strUserIDToDB, strPasswordToDB);
		GetAttachedClientSocket().WaitAll();
		return m_OpenRtn;
	}

	public bool BeginTrans()
	{
		BeginTransAsyn();
		GetAttachedClientSocket().WaitAll();
		return m_BeginTransRtn;
	}

	public bool ExecuteNoQuery(string strSQL)
	{
		ExecuteNoQueryAsyn(strSQL);
		GetAttachedClientSocket().WaitAll();
		return m_ExecuteNoQueryRtn;
	}

	public bool Commit(bool bSmart)
	{
		CommitAsyn(bSmart);
		GetAttachedClientSocket().WaitAll();
		return m_CommitRtn;
	}
}

File: SproInstall/SocketPro/tutorial/VBNet/MySecure/MySecureClient/MyBlowFish.vb
Imports SocketProAdapter
Imports SocketProAdapter.ClientSide
Imports USOCKETLib

Public Class CMySecure
	Inherits CAsyncServiceHandler
	Public Sub New(ByVal cs As CClientSocket, ByVal pDefaultAsyncResultsHandler As IAsyncResultsHandler)
		MyBase.New(MyBlowFishConst.sidCMySecure, cs, pDefaultAsyncResultsHandler)
	End Sub

	Public Sub New(ByVal cs As CClientSocket)
		MyBase.New(MyBlowFishConst.sidCMySecure, cs)
	End Sub

	Public Sub New()
		MyBase.New(MyBlowFishConst.sidCMySecure)
	End Sub

	Public m_nErrorCode As Integer = 0
	Public m_strErrorMessage As String

	Protected m_OpenRtn As String
	Protected Sub OpenAsyn(ByVal strUserIDToDB As String, ByVal strPasswordToDB As String)
		SendRequest(MyBlowFishConst.idOpenCMySecure, strUserIDToDB, strPasswordToDB)
	End Sub

	Protected m_BeginTransRtn As Boolean
	Public Sub BeginTransAsyn()
		SendRequest(MyBlowFishConst.idBeginTransCMySecure)
	End Sub

	Protected m_ExecuteNoQueryRtn As Boolean
	Public Sub ExecuteNoQueryAsyn(ByVal strSQL As String)
		SendRequest(MyBlowFishConst.idExecuteNoQueryCMySecure, strSQL)
	End Sub

	Protected m_CommitRtn As Boolean
	Public Sub CommitAsyn(ByVal bSmart As Boolean)
		SendRequest(MyBlowFishConst.idCommitCMySecure, bSmart)
	End Sub

	'When a result comes from a remote SocketPro server, the below virtual function will be called.
	'We always process returning results inside the function.
	Protected Overrides Sub OnResultReturned(ByVal sRequestID As Short, ByVal UQueue As CUQueue)
		UQueue.Pop(m_nErrorCode)
		UQueue.Load(m_strErrorMessage)

		Select Case sRequestID
		Case MyBlowFishConst.idOpenCMySecure
			UQueue.Load(m_OpenRtn)
		Case MyBlowFishConst.idBeginTransCMySecure
			UQueue.Pop(m_BeginTransRtn)
		Case MyBlowFishConst.idExecuteNoQueryCMySecure
			UQueue.Pop(m_ExecuteNoQueryRtn)
		Case MyBlowFishConst.idCommitCMySecure
			UQueue.Pop(m_CommitRtn)
		Case Else
		End Select
	End Sub
	Public Function Open(ByVal strUserIDToDB As String, ByVal strPasswordToDB As String) As String
		OpenAsyn(strUserIDToDB, strPasswordToDB)
		GetAttachedClientSocket().WaitAll()
		Return m_OpenRtn
	End Function

	Public Function BeginTrans() As Boolean
		BeginTransAsyn()
		GetAttachedClientSocket().WaitAll()
		Return m_BeginTransRtn
	End Function

	Public Function ExecuteNoQuery(ByVal strSQL As String) As Boolean
		ExecuteNoQueryAsyn(strSQL)
		GetAttachedClientSocket().WaitAll()
		Return m_ExecuteNoQueryRtn
	End Function

	Public Function Commit(ByVal bSmart As Boolean) As Boolean
		CommitAsyn(bSmart)
		GetAttachedClientSocket().WaitAll()
		Return m_CommitRtn
	End Function
End Class
